xds/internal/xdsclient/authority.go

```diff
@@ -495,7 +511,7 @@ func (a *authority) watchResource(rType xdsresource.Type, resourceName string, w
 			a.logger.Infof("Resource type %q with resource name %q found in cache: %s", rType.TypeName(), resourceName, state.cache.ToJSON())
 		}
 		resource := state.cache
-		a.serializer.TrySchedule(func(context.Context) { watcher.OnUpdate(resource) })
+		a.serializer.TrySchedule(func(context.Context) { watcher.OnUpdate(resource, xdsresource.NopDoneNotifier{}) })
```

Comment:
Why is this a no-op? Because it's already in cache?

Reply:
Right. And the ADS stream is not currently waiting for this update to be processed. Specifically, flow-control was not incremented for this resource/watcher.